src/RestrictedPython/transformer.py

```diff
@@ -1367,32 +1323,24 @@ def visit_arg(self, node):
         return self.node_contents_visit(node)
 
     def visit_Return(self, node):
-        """
-
-        """
+        """Allow `return` statements without restrictions."""
         return self.node_contents_visit(node)
 
     def visit_Yield(self, node):
         """Deny Yield unconditionally."""
         self.not_allowed(node)
 
     def visit_YieldFrom(self, node):
-        """
-
-        """
+        """All Async Functionality should be denied by default."""
         self.not_allowed(node)
 
     def visit_Global(self, node):
-        """
-
-        """
-        self.not_allowed(node)
+        """Make Global Statements allowed."""
+        return self.node_contents_visit(node)
 
     def visit_Nonlocal(self, node):
-        """
-
-        """
-        self.not_allowed(node)
+        """Allow Nonlocal Statements."""
+        return self.node_contents_visit(node)
```

Comment:
This is Python 3 only. There will be no code in the wild having `nonlocal` statements.
So I strongly tend to deny this statement until a security review.